keel-plugin/src/main/kotlin/com/netflix/spinnaker/keel/plugin/TaskLauncher.kt

```diff
@@ -93,6 +97,8 @@ class TaskLauncher(
       )
       .let {
         log.info("Started task {} to upsert {}", it.ref, subject)
+        // TODO: check this
+        publisher.publishEvent(TaskCreatedEvent(TaskRecord(taskId = it.taskId, taskName = description, resourceId = subject)))
```

Comment:
Do we gain anything by generating an event here as opposed to just calling the `taskTrackingRepository` directly to save the record?

Reply:
Yes - we want to have all the taskTrackingRepository logic in one place, which is in the OrcaAgent class.

Reply:
It's a pattern I've observed used widely in keel and other projects.